beacon-chain/sync/peerstatus/peer_statuses.go

```diff
@@ -12,30 +12,43 @@ import (
 )
 
 var lock sync.RWMutex
-var peerStatuses = make(map[peer.ID]*pb.Status)
-var lastUpdated = make(map[peer.ID]time.Time)
+var peerStatuses = make(map[peer.ID]*peerStatus)
+var failureCount = make(map[peer.ID]int)
+var maxFailureThreshold = 3
+
+type peerStatus struct {
+	status      *pb.Status
+	lastUpdated time.Time
+}
 
 // Get most recent status from peer in cache. Threadsafe.
 func Get(pid peer.ID) *pb.Status {
 	lock.RLock()
 	defer lock.RUnlock()
-	return peerStatuses[pid]
+	return peerStatuses[pid].status
```

Comment:
This could cause nil pointer errors

Reply:
the usage of this method is right after 
```
	for _, k := range peerstatus.Keys() {
```
if not we would have to return an error for this method. The same applies to the `LastUpdated` method.